src/libAtomVM/opcodesswitch.h

```diff
@@ -2466,7 +2469,12 @@ static const char *const out_of_memory_atom = "\xD" "out_of_memory";
 
                 TRACE("make_fun/2, fun_index=%i\n", fun_index);
                 #ifdef IMPL_EXECUTE_LOOP
-                    ctx->x[0] = make_fun(ctx, mod, fun_index);
+                    term f = make_fun(ctx, mod, fun_index);
+                    if (term_is_nil(f)) {
```

Comment:
term_is_invalid_term() might be a good choice when you are using term_invalid_term()

Reply:
Yeah, much better idea!